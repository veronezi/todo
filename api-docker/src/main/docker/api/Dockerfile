ARG VERSION
ARG POSTGRES_VERSION
FROM veronezi/todo-auth-keys:${VERSION} AS keys
FROM veronezi/sample-tomee:0.0.1-b10

COPY --chown=tomee:tomee postgresql.${POSTGRES_VERSION}.jar /opt/tomee/lib/

COPY --chown=tomee:tomee ROOT.war /opt/tomee/webapps/
COPY --chown=tomee:tomee tomcat-users.xml /opt/tomee/conf/
COPY --chown=tomee:tomee logging.properties /opt/tomee/conf/
COPY --chown=tomee:tomee tomee.xml /opt/tomee/conf/

RUN mkdir -p /opt/keys/public
ENV PUBLIC_KEY_DIR=/opt/keys/public
COPY --chown=tomee:tomee --from=keys /opt/keys/publickey.pem /opt/keys/public/todo_rsa.pub

WORKDIR /opt/tomee/

COPY --chown=tomee:tomee start.sh .
RUN chmod u+x start.sh