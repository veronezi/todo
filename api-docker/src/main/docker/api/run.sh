#!/bin/bash
set -e

IFS=', ' read -r -a array <<< "$WAIT_FOR"
for element in "${array[@]}"
do
    /usr/local/bin/dockerize -wait tcp://${element/@/:} -wait-retry-interval 5s -timeout 240s
done

echo "All dependencies are online. Starting up this service now."
export CATALINA_OPTS="$CATALINA_OPTS -DMyDatasource.JdbcUrl=jdbc:postgresql://$DB_URL/$DB_NAME -DMyDatasource.UserName=$DB_USR -DMyDatasource.Password=$DB_PWD"

/opt/tomee/bin/catalina.sh "$@"
