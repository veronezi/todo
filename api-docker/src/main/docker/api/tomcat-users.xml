<tomcat-users xmlns="http://tomcat.apache.org/xml"
              xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
              xsi:schemaLocation="http://tomcat.apache.org/xml tomcat-users.xsd"
              version="1.0">
    <role rolename="todo-user"/>
    <user username="jdoe" password="jdoe" roles="todo-user"/>
</tomcat-users>
