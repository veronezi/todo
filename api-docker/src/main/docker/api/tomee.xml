<?xml version="1.0" encoding="UTF-8"?>
<tomee>

    <Resource id="MyDatasource" type="DataSource">
        # Use CATALINA_OPTS to setup these variables
        # Example: -DMyDatasource.JdbcUrl=jdbc:postgresql://mydb:5432/mydbname -DMyDatasource.UserName=jdoe
        JdbcDriver   org.postgresql.Driver
        JdbcUrl      jdbc:postgresql://localhost/test
        UserName     postgres
        Password     pass

        testOnBorrow true
        validationQuery SELECT 1
        removeAbandoned true
    </Resource>

</tomee>
