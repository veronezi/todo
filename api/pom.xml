<?xml version="1.0" encoding="UTF-8"?>
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
         xsi:schemaLocation=" http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <parent>
        <groupId>com.veronezi</groupId>
        <artifactId>todo</artifactId>
        <version>0.0.1-SNAPSHOT</version>
    </parent>
    <artifactId>todo-api</artifactId>
    <packaging>war</packaging>
    <dependencies>
        <dependency>
            <groupId>org.apache.tomee</groupId>
            <artifactId>javaee-api</artifactId>
        </dependency>
        <dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
        </dependency>
        <dependency>
            <groupId>org.liquibase</groupId>
            <artifactId>liquibase-cdi</artifactId>
        </dependency>
        <dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.powermock</groupId>
            <artifactId>powermock-module-junit4</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.powermock</groupId>
            <artifactId>powermock-api-mockito</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-lang3</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.eclipse.microprofile.jwt</groupId>
            <artifactId>microprofile-jwt-auth-api</artifactId>
        </dependency>
        <dependency>
            <groupId>org.eclipse.microprofile.config</groupId>
            <artifactId>microprofile-config-api</artifactId>
        </dependency>
        <dependency>
            <groupId>org.apache.tomee</groupId>
            <artifactId>mp-jwt</artifactId>
        </dependency>
        <dependency>
            <groupId>commons-io</groupId>
            <artifactId>commons-io</artifactId>
        </dependency>
    </dependencies>
</project>

