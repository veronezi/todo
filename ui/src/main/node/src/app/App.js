import React from "react";
import {BrowserRouter, Route} from "react-router-dom";
import Login from "./Login";
import Navigator from "./Navigator";
import jss from "./App.jss";

const App = ({classes}) => (
    <BrowserRouter>
        <div className={classes.application}>
            <Route path="/login" component={Login}/>
            <Route path="/" component={Navigator}/>
        </div>
    </BrowserRouter>
);

export default jss(App);
