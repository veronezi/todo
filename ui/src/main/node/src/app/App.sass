.App
    display: flex
    justify-content: center
    align-items: stretch

.page
    width: 100%
    display: flex
    flex-direction: column