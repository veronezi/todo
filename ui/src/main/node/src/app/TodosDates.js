import React from "react";
import Typography from "@material-ui/core/Typography/Typography";
import jss from "./TodosDates.jss";

// we will mock this later to make tests not random
const functions = {
    getNow: () => new Date()
};

const options = {
    year: "numeric", month: "numeric", day: "numeric"
};

const TodosDates = ({classes}) => (
    <Typography className={classes.light} variant="subtitle1">
        {new Intl.DateTimeFormat("en-US", options).format(functions.getNow())}
    </Typography>
);

export default jss(TodosDates);
export {functions};
