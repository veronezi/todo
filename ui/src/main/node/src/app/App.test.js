import React from "react";
import App from "./App";
import store from "../store";
import {Provider} from "react-redux";
import api from "./Rest";
import * as sinon from "sinon";
import renderer from "react-test-renderer";
import {functions} from "./TodosDates";

let sandbox;

it("renders App without crashing", function () {
    const resolved = new Promise((r) => {
        return r({
            data: []
        })
    });
    sandbox.stub(api, "get").returns(resolved);
    sandbox.stub(functions, "getNow").returns(new Date("1995-12-17T03:24:00"));

    const component = renderer.create(
        <Provider store={store}><App/></Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

beforeAll(() => sandbox = sinon.createSandbox());

afterAll(() => sandbox.restore());
