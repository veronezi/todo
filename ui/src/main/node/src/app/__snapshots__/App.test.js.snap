// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App without crashing 1`] = `
<div
  className="App-application-0-2-1"
>
  <div
    className="withRouter(Navigator)-root-0-2-3"
  >
    <div
      className="Connect(withRouter(Todos))-page-0-2-14"
    >
      <div
        className="Connect(withRouter(Todos))-header-0-2-13"
      >
        <div
          className="Connect(withRouter(Todos))-title-0-2-9"
        >
          <div
            className="Connect(withRouter(Todos))-titleChild-0-2-10"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-1 Connect(withRouter(Todos))-light-0-2-8"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 17H4v2h10v-2zm6-8H4v2h16V9zM4 15h16v-2H4v2zM4 5v2h16V5H4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
            <h4
              className="MuiTypography-root-10 MuiTypography-h4-25 MuiTypography-colorPrimary-40 Connect(withRouter(Todos))-light-0-2-8"
            >
              Your Things
            </h4>
            <h6
              className="MuiTypography-root-10 MuiTypography-subtitle1-28 TodosDates-light-0-2-15"
            >
              12/17/1995
            </h6>
          </div>
          <div
            aria-valuenow={100}
            className="MuiLinearProgress-root-45 MuiLinearProgress-colorPrimary-46 MuiLinearProgress-determinate-48"
            role="progressbar"
          >
            <div
              className="MuiLinearProgress-bar-55 MuiLinearProgress-barColorPrimary-56 MuiLinearProgress-bar1Determinate-59"
              style={
                Object {
                  "transform": "scaleX(1)",
                }
              }
            />
          </div>
        </div>
        <div
          className="Connect(TodosNumbers)-numbers-0-2-18 Connect(TodosNumbers)-title-0-2-17"
        >
          <div
            className="Connect(TodosNumbers)-percentage-0-2-19"
          >
            <div
              aria-valuenow={100}
              className="MuiCircularProgress-root-63 MuiCircularProgress-colorPrimary-66 MuiCircularProgress-static-64"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "transform": "rotate(-90deg)",
                  "width": 40,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg-68"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle-69 MuiCircularProgress-circleStatic-70"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={
                    Object {
                      "strokeDasharray": "126.920",
                      "strokeDashoffset": "0.000px",
                    }
                  }
                />
              </svg>
            </div>
            <span
              className="MuiTypography-root-10 MuiTypography-caption-20 Connect(TodosNumbers)-value-0-2-16"
            >
              100
              % done
            </span>
          </div>
        </div>
      </div>
      <div
        className="Connect(withRouter(Todos))-content-0-2-11"
      >
        <div
          className="MuiGrid-container-73 MuiGrid-align-items-xs-center-81 MuiGrid-justify-xs-center-90"
        />
      </div>
      <button
        aria-label="Add"
        className="MuiButtonBase-root-180 MuiFab-root-170 MuiFab-primary-172 Connect(withRouter(Todos))-add-0-2-12"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiFab-label-171"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-1"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-183"
        />
      </button>
    </div>
  </div>
</div>
`;
